src/lib/models/useDeleteModelMutation.ts

```diff
@@ -1,26 +1,33 @@
-import { useDataEngine } from '@dhis2/app-runtime'
-import { useMutation } from 'react-query'
+import { FetchError, useDataEngine } from '@dhis2/app-runtime'
+import { useMutation, UseMutationOptions } from 'react-query'
+import { ImportSummary } from '../../types'
 import { Schema } from '../useLoadApp'
 
 type MutationFnArgs = {
     id: string
     displayName: string
-    messages?: string[]
 }
 
-type Options = Omit<Parameters<typeof useMutation>[1], 'mutationFn'>
+type DeleteMutationError = Omit<FetchError, 'details'> & {
+    details: ImportSummary
+}
+
+type Options = Omit<
+    UseMutationOptions<ImportSummary, DeleteMutationError, MutationFnArgs>,
+    'mutationFn'
+>
 
 export function useDeleteModelMutation(schema: Schema, options?: Options) {
     const engine = useDataEngine()
 
     return useMutation({
         ...options,
-        mutationFn: (variables: MutationFnArgs) => {
+        mutationFn: (variables) => {
             return engine.mutate({
                 resource: schema.plural,
                 id: variables.id,
                 type: 'delete',
-            })
+            }) as Promise<ImportSummary>
```

Comment:
We might want to have type-guard for usage of the error-object. Like `isErrorReports()`, because this isn't exactly "safe".